File: _src/main/java/raccoonlang/tokenizer/Token.java
package raccoonlang.tokenizer;

public class Token {

    private TokenType tokenType;
    private int line;
    private int column;
    private String text;
    private String fileName;
    private boolean skip;

    public Token(TokenType tokenType, int line, int column, String text, String fileName) {
        this(tokenType, line, column, text, fileName, false);
    }

    public Token(TokenType tokenType, int line, int column, String text, String fileName, boolean skip){
        this.tokenType = tokenType;
        this.line = line;
        this.column = column;
        this.text = text;
        this.fileName = fileName;
        this.skip = skip;
    }

    public TokenType getTokenType() {
        return tokenType;
    }

    public int getLine() {
        return line;
    }

    public int getColumn() {
        return column;
    }

    public String getText() {
        return text;
    }

    @Override
    public String toString() {
        return "Token{" +
                "tokenType=" + tokenType +
                ", line=" + line +
                ", column=" + column +
                ", text='" + text + '\'' +
                '}';
    }

    public String getFileName() {
        return fileName;
    }

    public boolean isSkip() {
        return skip;
    }
}

File: Tokenizing/Token.cs
namespace Raccoonlang.Tokenizing;

public class Token
{
    public TokenType Type {get; private set;}
    public int Line {get; private set;}
    public int Column {get; private set;}
    public string Text {get; private set;}
    public string FileName {get; private set;}
    public bool Skip {get; private set;}

    public Token(TokenType type, int line, int column, string text, string fileName)
    {
        this.Type = type;
        this.Line = line;
        this.Column = column;
        this.Text = text;
        this.FileName = fileName;
        this.Skip = false;
    }
    
    public Token(TokenType type, int line, int column, string text, string fileName, bool skip)
    {
        this.Type = type;
        this.Line = line;
        this.Column = column;
        this.Text = text;
        this.FileName = fileName;
        this.Skip = skip;
    }
}